website/content/plugins/drivers/community/pledge.mdx

```diff
@@ -0,0 +1,333 @@
+enable Nomad tasks using the pledge driver to bind to privilged ports (i.e.
+below 1024) while using host networking mode.
+
+```shell-session
+sudo setcap cap_net_bind_service+eip /opt/bin/pledge-1.8.com
+```
+
+For convenience the `driver.pledge.cap.net_bind` Client attribute will indicate
+whether the Linux capability is set on the executable.
+
+#### bridge networking
+
+If using `bridge` networking mode, be sure to install the standard [CNI plugins]
+[cni] as required by the Nomad client.
+
+## Client Requirements
+
+The `pledge` driver requires the following:
+
+- 64-bit amd64 Linux host
+  - kernel version 5.13+
+  - cgroups v2 enabled
+- The `pledge` driver binary placed in [plugin_dir][plugin_dir]
+- The `pledge.com` utility executable on the host
+
+## Plugin Options
+
+- `pledge_executable` - The path to the `pledge.com` binary. The pledge utility
+can be [downloaded][download] directly from its creator, or compiled manually
+from [source][pledge_github].
+
+```hcl
+plugin "nomad-pledge-driver" {
+  config {
+    pledge_executable = "/opt/bin/pledge-1.8.com"
+  }
+}
+```
+
+## Client Attributes
+
+The `pledge` driver will set the following Client attributes:
+
+| Attribute | Description |
+| ----------| ----------- |
+| `driver.pledge` | set to `1` if the driver is enabled |
+| `driver.pledge.abs` | the absolute path to the pledge.com utility |
+| `driver.pledge.cap.net_bind` | indicates whether the NET_BIND linux capability is set |
+| `driver.pledge.os` | the detected operating system |
+
+## Resource Isolation
+
+CPU and memory resource isolation is enforced by cgroups v2 features, and is
+configured through the resources block in task configuration.
+
+```hcl
+resources {
+  cores  = 3
+  memory = 512
+}
+```
+
+#### cpu bandwidth
+
+The `pledge` driver always enforces a maximum CPU bandwidth available to the
+task. The bandwidth is calculated whether `resources.cpu` or `resources.cores`
+is configured.
+
+If `resources.cores` is set, the scheduler reserves that many CPU cores for use
+by the task, and only this task will be able to run on those cores.
+
+#### memory oversubscription
+
+If the Nomad scheduler is configured to enable memory [oversubscription]
+[oversub], the `pledge` driver will correctly enable configuring `memory_max`
+in addition to `memory`. In this case, `memory_max` indicates the maximum amount
+of memory the task is able to request before being OOM killed, and `memory`
+represents a minimum amount of memory the kernel will gauruntee is available for
+the Task.
+
+```hcl
+resources {
+  cpu = 2000
+  memory = 1024
+  memory_max = 2048
+}
+```
+
+## Capabilities
+
+The `pledge` driver implements the following [capabilities][capabilities].
+
+| Feature              | Implementation    |
+| ------------         | ------------------|
+| send signals         | true              |
+| exec                 | false             |
+| filesystem isolation | Landlock LSM      |
```

Comment:
Did we end up settling on whether we were going to have a separate isolation capability for this?

Reply:
ooh good catch; yeah we're just going to piggyback off of `None` until we have a reason to differentiate on landlock on the client side (probably when mounts come into play, or something)